sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/externalrules/CxxExternalRulesSensor.java

```diff
@@ -59,11 +72,30 @@ protected String reportPathKey() {
   public void describe(SensorDescriptor descriptor) {
     descriptor.onlyOnLanguage(CxxLanguage.KEY).name("CxxExternalRulesSensor");
   }
-  
+
+  /**
+   * {@inheritDoc}
+   */
   @Override
-  public void processReport(final SensorContext context, File report) throws XMLStreamException {
+  public void execute(SensorContext context) {
+    try {
+      transformFiles();
+    } catch (Exception e) {
```

Comment:
Reasons to catch not only on highest level:
* Could be tool A needs transforming A and tool B needs transformation B. If A is failing B should nevertheless work (tolerant mode)
* Same if more then one file has to be transformed. If transformation A handles more then one file it should continue if one transformation is failing. 

> So you suggest different behavior for both modes? ;-)

Yes behavior of all sensors is different depending on setting of 'sonar.cxx.errorRecoveryEnabled'. But all what you have to do is to call:
https://github.com/SonarOpenCommunity/sonar-cxx/blob/master/sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/utils/CxxUtils.java#L88